stl/src/winapinls.cpp

```diff
@@ -5,10 +5,495 @@
 
 #if _STL_WIN32_WINNT < _WIN32_WINNT_VISTA
 
-#include "nlsdownlevel.h"
 #include <ctype.h>
 #include <stdlib.h>
 
+namespace {
+struct LCIDTOLOCALENAME {
+   LCID lcid;
+   const wchar_t* localeName;
+};
+
+
+struct LOCALENAMEINDEX {
+   const wchar_t* name;
+   int index;
+};
+
+
+// Map of LCID to locale name for Windows XP.
+// Data in this table has been obtained from National Language Support (NLS) API Reference.
+// The table is sorted to improve search performance.
+// clang-format off
+const LCIDTOLOCALENAME LcidToLocaleNameTable[] = {

```

Comment:
Now that this is touched should it become `constexpr LCIDTOLOCALENAME LcidToLocaleNameTable[]` or even better `std::array<LCIDTOLOCALENAME , MeToLazyToCount> LcidToLocaleNameTable`?

Reply:
Good point regarding `std::array` and `constexpr`. I figured to keep changes minimal in changing the actual structures. I'll adapt the code.

Given that all functions are marked with `extern "C"` I guess they are all actually exported? So no changes there?

With regard to `static` is this the preferred way to have internal linkage or is the use of anonymous namespace accepted?

Reply:
@chris0x44 maybe wait for a maintainer before changing stuff ;)

Reply:
* Changing builtin array to `std::array` here seems like unnecessary churn.
* `constexpr` is at least borderline useful, since it will prevent someone from inadvertently changing `LCIDTOLOCALENAME`'s constructor in a way that would cause this array to need dynamic initialization.
* Between the unnamed namespace and being objects of `const`-qualified type, these already have internal linkage for two reasons - I see no need to add a third.

Reply:
When moving code between files (unlike whole-file renames), it is mechanically difficult to diff additional changes (it can be done, it's just not automatic/convenient), so I strongly recommend against making significant changes in the same PR unless it is absolutely necessary.

Changing `const` to `constexpr` here is a very safe change, so doing that in the same PR is fine.